controllers/crypto.js

```diff
@@ -0,0 +1,50 @@
+const { fetchProducts, fetchProduct } = require('../models/crypto')
+
+const ERROR_MESSAGE = 'Something went wrong. Please try again or contact admin'
+
+/**
+ * Get the  products details
+ * @param {Object} req - Express request object
+ * @param {Object} res - Express response object
+ */
+
+const getProducts = async (req, res) => {
+  try {
+    const productData = await fetchProducts()
+    return res.json({
+      message: Object.keys(productData).length !== 0 ? 'Products returned successfully!' : 'No products found',
+      products: productData
+    })
+  } catch (err) {
+    logger.error(`Error while retriving products ${err}`)
+    return res.boom.badImplementation(ERROR_MESSAGE)
+  }
+}
+
+/**
+ * Get the  product details
+ * @param {Object} req - Express request object
+ * @param {Object} res - Express response object
+ */
+
+const getProduct = async (req, res) => {
+  try {
+    const productId = req.params.productId
```

Comment:
joi validation missing

Reply:
This is GET call to product/:productid, is validation necessary here?